packages/astro/src/server/middleware.ts

```diff
@@ -184,12 +184,18 @@ async function instrumentRequest(
 
               const newResponseStream = new ReadableStream({
                 start: async controller => {
-                  for await (const chunk of originalBody) {
-                    const html = typeof chunk === 'string' ? chunk : decoder.decode(chunk, { stream: true });
-                    const modifiedHtml = addMetaTagToHead(html);
-                    controller.enqueue(new TextEncoder().encode(modifiedHtml));
+                  try {
+                    for await (const chunk of originalBody) {
+                      const html = typeof chunk === 'string' ? chunk : decoder.decode(chunk, { stream: true });
+                      const modifiedHtml = addMetaTagToHead(html);
+                      controller.enqueue(new TextEncoder().encode(modifiedHtml));
+                    }
+                  } catch (e) {
+                    sendErrorToSentry(e);
+                    controller.error(e);
```

Comment:
I think we should still rethrow this error, otherwise we risk breaking people who rely on this.

Is `controller.error(e);` robust enough?

Reply:
`controller.error` is what rethrows the error on a stream. Throwing from this function is ignored (it was being ignored before, so no one was relying on the error working).

The test is checking that the error is being forwarded.

Reply:
Hmm I'm not sure if we want to capture this error at all. The only reason we're even going through the body is to inject our `<meta>` tags and if it's _us_ causing the error by doing so, we [shouldn't report it](https://develop.sentry.dev/sdk/philosophy/#degrade-gracefully). 

Or would we even swallow errors if we didn't modify the HTML?

The `try/catch` is probably a good idea anyway to degrade gracefully 🤔

Reply:
> Hmm I'm not sure if we want to capture this error at all.

This is a user thrown error that is being missed and shows as `Internal Server Error` on rendering. I think it makes sense for Sentry to capture and report it, but most of all it makes sense to not drop it and pass it forward so errors from the user are not discarded.

We could have some more verbose code to handle only errors from the iterator and not from any part of the `<meta>` tag inside the loop.

Reply:
Discussed this with @AbhiPrasad offline and for now we think it's fine to leave as-is. The injection is fairly low risk (famous last words lol). In case we get reports about our HTML injection logic causing errors, we can still revisit. 

Thanks for contributing!